storage/interface.go

```diff
@@ -28,9 +28,10 @@ import (
 // The errors exposed.
 var (
 	ErrNotFound                    = errors.New("not found")
-	ErrOutOfOrderSample            = errors.New("out of order sample")
-	ErrDuplicateSampleForTimestamp = errors.New("duplicate sample for timestamp")
-	ErrOutOfBounds                 = errors.New("out of bounds")
+	ErrOutOfOrderSample            = errors.New("out of order sample")            // OOO support disabled and sample is OOO
+	ErrTooOldSample                = errors.New("too old sample")                 // OOO support enabled, but sample outside of tolerance
```

Comment:
WDYT @codesome ?

I remember we introduced this new error because we wanted to explicitely make the distinction between real out-of-order when out-of-order is disabled and discarded out-of-order samples because they are older than the configured window when out-of-order was enabled.

I think originally we were concerned that the user would be confused by out-of-order errors when the feature was enabled. Maybe we could simplify this?